src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -199,6 +200,7 @@ def _get_azure_cli_properties(self):
         set_custom_properties(result, 'PythonVersion', platform.python_version())
         set_custom_properties(result, 'ModuleCorrelation', self.module_correlation)
         set_custom_properties(result, 'ExtensionName', ext_info)
+        set_custom_properties(result, 'CLIRecommendation', self.cli_recommendation)
```

Comment:
BTW, have tested if it can be serialized successfully with json format? I remember there could be serialization failure with no-string value